v2/pkg/genruntime/kubernetes_resource.go

```diff
@@ -91,3 +93,17 @@ func GetAPIVersion(metaObject ARMMetaObject, scheme *runtime.Scheme) (string, er
 
 	return rsrc.GetAPIVersion(), nil
 }
+
+// GetResourceTypeAndProvider returns the provider and the array of resource types which represent the resource.
+// For example: Microsoft.Compute/virtualMachineScaleSets would return ("Microsoft.Compute", []string{"virtualMachineScaleSets"}, nil)
+func GetResourceTypeAndProvider(res ARMMetaObject) (string, []string, error) {
```

Comment:
I think we already have this code somewhere. We should either reuse the original, or change the other case to use this one. Maybe as a separate PR though.

Reply:
Not sure if you saw but I just moved it from where it was elsewhere. Previously it was unexpected in `resource_hierarchy.go`. Unless you're saying it's _already_ in two different places in the code? One in `resource_hierarchy.go` and one someplace else?